### PR TITLE
TOMEE-2988: Xbean 4.18 + asm9

### PR DESCRIPTION
**Pull request by cocorossello**

### CODE REVIEW

**Review by Daniel-Dos — APPROVED** [review]
very good for me .

**Comment by rzo1** [comment]
@cesarhernandezgt Do you have an idea, why we lost the Jenkins PR builder checks for (newer) pull requests?

File: arquillian/arquillian-tomee-common/pom.xml [hunk 1]
```diff
@@ -86,6 +86,16 @@
       <artifactId>xbean-finder-shaded</artifactId>
       <version>${xbeanVersion}</version>
       <scope>provided</scope>
+      <exclusions>
```
Comment (rzo1) [inline arquillian/arquillian-tomee-common/pom.xml:89]: I am wondering:

Is this needed?  Afaik [xbean-finder-shaded](https://mvnrepository.com/artifact/org.apache.xbean/xbean-finder-shaded/4.18) only has a compile dependency towards `xbean-asm9-shaded`. The `org.ow2.asm` stuff is a transient dependency of `xbean-asm9-shaded`. So shouldn't we exclude  `xbean-asm9-shaded` here as it comes from `openwebbeans` anyway?

File: container/openejb-core/pom.xml [hunk 2]
```diff
@@ -562,11 +562,31 @@
...
+        <exclusion>
+          <groupId>org.ow2.asm</groupId>
+          <artifactId>asm-commons</artifactId>
+        </exclusion>
+      </exclusions>
     </dependency>
     <dependency>
       <groupId>org.apache.xbean</groupId>
       <artifactId>xbean-finder-shaded</artifactId>
+      <exclusions>
```
Comment (rzo1) [inline container/openejb-core/pom.xml:580]: see above

File: arquillian/arquillian-tomee-common/pom.xml (hunk 1 above)
Comment (rmannibucau) [reply arquillian/arquillian-tomee-common/pom.xml:89]: ow2 deps should be excluded in root pom in dependency management of xbean-asm9-shaded dependency declaration to ensure the whole tree gets it for free.

Comment (cocorossello) [reply arquillian/arquillian-tomee-common/pom.xml:89]: Ok, let me change it tomorrow. Thanks for the hint, didn't really now where to exclude or if all exclusions were needed

Comment (cocorossello) [reply arquillian/arquillian-tomee-common/pom.xml]: Changed it, only that exclusion is enough

**Review by rzo1 — APPROVED** [review]

**Review by mawiesne — APPROVED** [review]

**Pull request merged**